go-controller/pkg/ovn/pods.go

```diff
@@ -231,16 +300,25 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) {
 			logrus.Errorf("Error while creating logical port %s "+
 				"stdout: %q, stderr: %q (%v)",
 				portName, out, stderr, err)
-			return
+			return ""
 		}
 	}
 
 	oc.logicalPortCache[portName] = logicalSwitch
 
 	gatewayIP, mask, err := oc.getGatewayFromSwitch(logicalSwitch)
```

Comment:
do you need to run getGateway for the non default port?

Reply:
Yes, it may have gateway. But unlike the default one, it may also have no gateway.